### PR TITLE
REM-25613 Send HTTP/S network request status information to backend.

### PR DESCRIPTION
**Pull request by vito1188**

### CODE REVIEW

File: RPerformanceTracking/Private/_RPTMeasurement.h [hunk 1]
```diff
@@ -21,6 +21,8 @@ RPT_EXPORT @interface _RPTMeasurement : NSObject
 @property (nonatomic, nullable)       NSObject           *receiver;
 @property (nonatomic, copy, nullable) NSString           *method;
 @property (nonatomic, copy, nullable) NSString           *screen;
+@property (nonatomic)                 NSUInteger          statusCode;
```
Comment (donnie-jp) [inline RPerformanceTracking/Private/_RPTMeasurement.h]: according to docs this should be an `NSInteger`

File: RPerformanceTracking/Private/_RPTTracker.m [hunk 2]
```diff
@@ -102,6 +102,16 @@ - (void)end:(uint_fast64_t)trackingIdentifier
     }
 }
 
+- (void)end:(uint_fast64_t)trackingIdentifier statusCode:(NSInteger)statusCode
```
Comment (donnie-jp) [inline RPerformanceTracking/Private/_RPTTracker.m]: Can we get the other version of this method (without statusCode) to just call this with a 0 statusCode param?

File: RPerformanceTracking/Private/_RPTClassManipulator+NSURLSessionTask.m [hunk 3]
```diff
@@ -53,7 +53,20 @@ void _handleChangedState(NSURLSessionTask *task, NSURLSessionTaskState state)
...
-            [_RPTTrackingManager.sharedInstance.tracker end:trackingIdentifier];
+            NSInteger statusCode = 0;
+            if ([task.response isKindOfClass:[NSHTTPURLResponse class]])
+            {
+                NSHTTPURLResponse *httpResponse = (NSHTTPURLResponse *)task.response;
+                statusCode = httpResponse.statusCode;
+            }
+            if (statusCode)
+            {
+                [_RPTTrackingManager.sharedInstance.tracker end:trackingIdentifier statusCode:statusCode];
```
Comment (donnie-jp) [inline RPerformanceTracking/Private/_RPTClassManipulator+NSURLSessionTask.m]: I don't think we need the `if (statusCode)` check and we can just always call this unconditionally. 
Even if statusCode is zero (i.e. `statusCode = httpResponse.statusCode` wasn't hit) it won't change the outcome cos zero is the initial value of `measurement.statusCode`.

**Review by donnie-jp — CHANGES_REQUESTED** [review]
A few small changes required.

File: RPerformanceTracking/Private/_RPTClassManipulator+NSURLSessionTask.m (hunk 3 above)
Comment (vito1188) [reply RPerformanceTracking/Private/_RPTClassManipulator+NSURLSessionTask.m]: yes.

**Review by donnie-jp — APPROVED** [review]

**Pull request merged**